File: server_pub.js
const config = require('config');
const SERVER_CFG = config.get("SERVER");
const ORIGIN = SERVER_CFG.origin;

const express = require("express");
const morgan = require("morgan");
const fs = require("fs");
const path = require("path");
const cors = require("cors");
let logger = require("./logger");

var app = express();

app.use(cors());

app.use(express.json());
app.use(express.urlencoded({
    extended: true
}));

// app.use(express.static('public'));

// logger
let successLogStream = fs.createWriteStream(path.join(__dirname, "logs/http_success.log"), {
    flags: 'a'
})
let errorLogStream = fs.createWriteStream(path.join(__dirname, "logs/http_error.log"), {
    flags: 'a'
});

// setup the logger middleware
app.use([morgan('combined', {
        skip: function (req, res) {
            return (res.statusCode <= 599 && res.statusCode >= 400)
        },
        stream: successLogStream
    }),
    morgan('combined', {
        skip: function (req, res) {
            return (res.statusCode <= 399 && res.statusCode >= 100)
        },
        stream: errorLogStream
    })

]);

app.use(morgan('dev'));

// ROUTES
require("./routes/dev")(app);

app.listen(SERVER_CFG.api.Publisher_API_PORT, "127.0.0.1", 511, () => {
    logger.info(`Publisher server is running on port ${SERVER_CFG.api.Publisher_API_PORT}`)
});
app.runningPort = SERVER_CFG.api.Publisher_API_PORT

File: server_pub.py
import logging
from logging.handlers import TimedRotatingFileHandler
import os
import argparse

from Configs import baseConfig

config = baseConfig()
api = config["API"]

parser = argparse.ArgumentParser()
parser.add_argument("--logs", help="Set log level")
args = parser.parse_args()

log_level = args.logs or "info"
numeric_level = getattr(logging, log_level.upper(), None)

if not isinstance(numeric_level, int):
    raise ValueError("Invalid log level: %s" % log_level)

if not os.path.exists("logs/"):
    os.makedirs("logs/")

logging.basicConfig(level=numeric_level)

logger = logging.getLogger()
rotatory_handler = TimedRotatingFileHandler(
    "logs/combined.log", when="D", interval=1, backupCount=30
)
rotatory_handler.setLevel(logging.INFO)
formatter = logging.Formatter(
    "%(asctime)s | %(levelname)s | %(message)s", "%m-%d-%Y %H:%M:%S"
)
rotatory_handler.setFormatter(formatter)
logger.addHandler(rotatory_handler)

from flask import Flask

from routes.publisher.v2 import v2

app = Flask(__name__)

app.register_blueprint(v2, url_prefix="/v2")

if __name__ == "__main__":
    app.logger.info("Running on un-secure port: %s" % api['PUBLISHER_PORT'])
    app.run(host=api["HOST"], port=api["PUBLISHER_PORT"])